src/internal/pfsdb/branches.go

```diff
@@ -731,6 +748,21 @@ func DeleteBranchTrigger(ctx context.Context, tx *pachsql.Tx, from BranchID) err
 	return nil
 }
 
+func GetBranchPropagationSpecDirectSubvenanceRows(ctx context.Context, ext sqlx.ExtContext, id BranchID) ([]*BranchPropagationSpecRow, error) {
+	var rows []*BranchPropagationSpecRow
+	if err := sqlx.SelectContext(ctx, ext, &rows, `
+		SELECT bp.from_id, bp.to_id, CASE WHEN never IS NULL THEN false ELSE never END 
```

Comment:
[nit] you may want to call 'never' 'bps.never' to make it clear which table the column comes from.

Reply:
Hmm, to me it seems cleaner to not use the table namespacing unless it is needed.